source/opt/inline_pass.h

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) 2017 The Khronos Group Inc.
+// Copyright (c) 2017 Valve Corporation
+// Copyright (c) 2017 LunarG Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef LIBSPIRV_OPT_INLINE_PASS_H_
+#define LIBSPIRV_OPT_INLINE_PASS_H_
+
+#include <algorithm>
+#include <memory>
+#include <unordered_map>
+#include <vector>
+
+#include "def_use_manager.h"
+#include "module.h"
+#include "pass.h"
+
+namespace spvtools {
+namespace opt {
+
+// See optimizer.hpp for documentation.
+class InlinePass : public Pass {
+ public:
+  InlinePass();
+  Status Process(ir::Module*) override;
+
+  const char* name() const override { return "inline"; }
+
+ private:
+  // Return the next available Id and increment it.
+  inline uint32_t TakeNextId() { return next_id_++; }
+
+  // Find pointer to type and storage in module, return its resultId,
+  // 0 if not found. TODO(greg-lunarg): Move this into type manager.
+  uint32_t FindPointerToType(uint32_t type_id, uint32_t storage_id);
```

Comment:
The "storage_id" parameter is actually being used as the literal value for StorageClass.  It's not an Id.  So the name is misleading.   

Additionally, you should tighten the type to SpvStorageClass.  Then it would be hard to misunderstand or use incorrectly.  That comes at the cost of some casting, but I think that's probably worth it.